pkg/metadata/sandbox_test.go

```diff
@@ -0,0 +1,127 @@
+limitations under the License.
+*/
+
+package metadata
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	assertlib "github.com/stretchr/testify/assert"
+
+	"github.com/kubernetes-incubator/cri-containerd/pkg/metadata/store"
+
+	"k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
+)
+
+func TestSandboxMetadata(t *testing.T) {
+	testID := "TestID"
+	testMetadata := SandboxMetadata{
+		ID:   testID,
+		Name: "TestName",
+		Config: &runtime.PodSandboxConfig{
+			Metadata: &runtime.PodSandboxMetadata{
+				Name:      "TestPod",
+				Uid:       "TestUid",
+				Namespace: "TestNamespace",
+				Attempt:   1,
+			},
+		},
+		CreatedAt: time.Now().UnixNano(),
+		NetNS:     "TestNetNS",
+	}
+	assert := assertlib.New(t)
+
+	s := NewSandboxStore(store.NewMetadataStore())
+
+	t.Logf("should be able to create sandbox metadata")
+	assert.NoError(s.Create(testMetadata))
+	sb, err := s.Get(testID)
+	assert.NoError(err)
+
+	t.Logf("should be able to get sandbox metadata")
+	oldMetadata, err := sb.GetMetadata()
+	assert.NoError(err)
+	assert.Equal(testMetadata, oldMetadata)
+
+	t.Logf("failed update should not take effect")
+	updateErr := errors.New("update error")
+	newCreatedAt := time.Now().UnixNano()
+	err = sb.Update(func(o SandboxMetadata) (SandboxMetadata, error) {
+		o.CreatedAt = newCreatedAt
+		return o, updateErr
+	})
+	assert.Equal(updateErr, err)
+	newMetadata, err := sb.GetMetadata()
+	assert.NoError(err)
+	assert.Equal(testMetadata, newMetadata)
+
+	t.Logf("successful update should take effect")
+	expectMetadata := testMetadata
+	expectMetadata.CreatedAt = newCreatedAt
+	err = sb.Update(func(o SandboxMetadata) (SandboxMetadata, error) {
+		o.CreatedAt = newCreatedAt
+		return o, nil
+	})
+	assert.NoError(err)
+	newMetadata, err = sb.GetMetadata()
+	assert.NoError(err)
+	assert.Equal(expectMetadata, newMetadata)
+
+	t.Logf("update should not affect existing snapshot")
+	assert.Equal(testMetadata, oldMetadata)
+}
+
+func TestSandboxStore(t *testing.T) {
+	sandboxes := map[string]SandboxMetadata{
+		"1": {ID: "1", Name: "sandbox-1"},
+		"2": {ID: "2", Name: "sandbox-2"},
+		"3": {ID: "3", Name: "sandbox-3"},
+	}
+	assert := assertlib.New(t)
+
+	s := NewSandboxStore(store.NewMetadataStore())
+
+	t.Logf("should be able to create sandbox metadata")
+	for _, meta := range sandboxes {
+		assert.NoError(s.Create(meta))
+	}
+
+	t.Logf("should be able to get sandbox metadata")
+	for id, expectMeta := range sandboxes {
+		sb, err := s.Get(id)
+		assert.NoError(err)
+		meta, err := sb.GetMetadata()
+		assert.NoError(err)
+		assert.Equal(expectMeta, meta)
+	}
+
+	t.Logf("should be able to list sandbox metadata")
+	sbs, err := s.List()
+	assert.NoError(err)
+	assert.Len(sbs, 3)
+
+	t.Logf("should be able to delete sandbox metadata")
+	testID := "2"
+	assert.NoError(s.Delete(testID))
+	sbs, err = s.List()
+	assert.NoError(err)
+	assert.Len(sbs, 2)
+	meta, err := s.Get(testID)
+	assert.NoError(err)
+	assert.True(meta == nil)
+}
```

Comment:
I'd add another test that creates the metadata again for id..  is that the expected pattern? or are we only deleting the metadata as we delete a sandbox?